File: Famoser.MassPass.Data/Entities/Communications/Response/Base/ApiResponse.cs
using System;
using Famoser.FrameworkEssentials.Attributes;
using Famoser.FrameworkEssentials.Helpers;
using Famoser.FrameworkEssentials.Models.RestService;
using Famoser.MassPass.Data.Enum;

namespace Famoser.MassPass.Data.Entities.Communications.Response.Base
{
    public class ApiResponse
    {
        public ApiResponse()
        {
            ApiError = ApiError.None;
        }

        public ApiError ApiError { get; set; }
        public string ApiMessage { get; set; }
        public HttpResponseModel RawResponse { get; set; }
        public bool IsSuccessfull => RawResponse != null && RawResponse.IsRequestSuccessfull && ApiError == ApiError.None;
        public string GetApiErrorAsString => ReflectionHelper.GetAttributeOfEnum<DescriptionAttribute, ApiError>(ApiError).Description;
    }
}

File: Famoser.MassPass.Api/src/Models/Response/Base/ApiResponse.php
<?php
/**
 * Created by PhpStorm.
 * User: famoser
 * Date: 23.05.2016
 * Time: 09:41
 */

namespace Famoser\MassPass\Models\Response\Base;


use Famoser\MassPass\Types\ApiErrorTypes;

class ApiResponse
{
    public function __construct($successfull = true, $apiError = ApiErrorTypes::None)
    {
        $this->Successfull = $successfull;
        $this->ApiError = $apiError;
    }

    public $Successfull;
    public $ApiError;
    public $ApiMessage;
}